### PR TITLE
alarm setting

### PR DESCRIPTION
**Pull request by SophieYoonseo**
After setting the alarm, I noticed a push notification popping up on the phone. When I tap on push notifications, an alarm-shaped image pops up. But neither vibration nor sound. So I set it to vibrate until I pressed 'DISMISS'.

### CODE REVIEW

**Comment by PPartisan** [comment]
Thanks Sophie. I'll look at this over the weekend.

**Comment by SophieYoonseo** [comment]
@PPartisan Okay, thanks! I will wait for your answer. I would like a quick review.

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 1]
```diff
@@ -1,45 +1,120 @@
...
+import android.widget.TextView;
 
 import com.github.ppartisan.simplealarms.R;
 
-public final class AlarmLandingPageFragment extends Fragment implements View.OnClickListener {
 
+import static android.content.Context.VIBRATOR_SERVICE;
+
+public final class AlarmLandingPageFragment extends Fragment implements SensorEventListener {
+    public  long lastTime;
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: These scope of these can be reduced - `private` is fine.

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 2]
```diff
@@ -1,45 +1,120 @@
...
+    public  float speed;
+    public  float lastX;
+    public  float lastY;
+    public  float lastZ;
+    public  float x, y, z;
+    private int count;
+
+
+    public static  int SHAKE_THRESHOLD = 100;
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: These are constants, so can be `final`

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 3]
```diff
@@ -1,45 +1,120 @@
...
-        dismiss.setOnClickListener(this);
 
+        startvibe();
         return v;
     }
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (sensorManager != null)
+            sensorManager.unregisterListener(this);
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: Where do you set this listener?

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 4]
```diff
@@ -1,45 +1,120 @@
...
+        });
+        dismiss.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                vibrator.cancel();
+                getActivity().finish();
+            }
+        });
+        pm=(PowerManager)getActivity().getSystemService(Context.POWER_SERVICE);
+        //wl=pm.newWakeLock(PowerManager.SCREEN_BRIGHT_WAKE_LOCK,"mytag");
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: Delete this if it's unused

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 5]
```diff
@@ -1,45 +1,120 @@
...
+    public void onSensorChanged(SensorEvent event) {
+        if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+            long currentTime = System.currentTimeMillis();
+            long gabOfTime = (currentTime - lastTime);
+            if (gabOfTime > 100) {
+                lastTime = currentTime;
+                x = event.values[SensorManager.DATA_X];
+                y = event.values[SensorManager.DATA_Y];
+                z = event.values[SensorManager.DATA_Z];
+                speed = Math.abs(x + y + z - lastX - lastY - lastZ) / gabOfTime * 10000;
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: What is the significance of the `10000` here?

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java [hunk 6]
```diff
@@ -1,45 +1,120 @@
...
+                if (speed > SHAKE_THRESHOLD) {
+                    lastTime=currentTime;
+                    count++;//for shake test
+                    if(count==10)
```
Comment (PPartisan) [inline app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: Not for this PR, but after I've merged in your work I will probably make these optional, configurable features (vibrating main activity, shake to dismiss, vibration pattern, number of shakes required to dismiss, etc.)

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 1 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan  I fixed it

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 2 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan I fixed it

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 3 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan I added an api named 'sensorManager' and called 'unregisterListener' in it.
It is detailed in the link below.
https://developer.android.com/reference/android/hardware/SensorManager

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 4 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan I fixed it.

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 5 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan If you want to turn off the alarm by setting a strong shaking strength when turning off the alarm, you should lower the number. Conversely, if you want to set the shaking strength weakly, increase the number. I randomly set it to 10000.

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 6 above)
Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan It's okay to tweak the functionality a bit to suit your specific characteristics.

File: app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java (hunk 3 above)
Comment (PPartisan) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: I understand - but don't you also need to call `sensorManager#registerListener` somewhere as well (presumably inside `onStart()`)?

Comment (SophieYoonseo) [reply app/src/main/java/com/github/ppartisan/simplealarms/ui/AlarmLandingPageFragment.java]: @PPartisan Ok! I understand! I added `onStart()`. You can turn off the alarm by shaking it. If you want, you can tweak some of features you mentioned (vibrating main activity, shake to dismiss, vibration pattern, number of shakes required to dismiss, etc.)

**Comment by PPartisan** [comment]
Thanks for your changes. I'll manually test this work tonight or tomorrow evening to confirm it behaves as it's supposed to, and if all goes well I'll merge afterwards.

**Comment by SophieYoonseo** [comment]
Ok! I’m waiting for your answer!!

**Comment by SophieYoonseo** [comment]
@PPartisan I am looking forward to your answer. When I tried, it was no problem. I don't have so much time right now. If there is a problem, I will fix it. If not, I would like you to merge.

**Pull request merged**